lib/Service/EncryptService.php

```diff
@@ -0,0 +1,272 @@
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+namespace OCA\Passman\Service;
+
+
+// Class copied from http://stackoverflow.com/questions/5089841/two-way-encryption-i-need-to-store-passwords-that-can-be-retrieved?answertab=votes#tab-top
+// Upgraded to use openssl
+class EncryptService {
+
+	/**
+	 * Supported cipher algorithms accompanied by their key/block sizes in bytes
+	 *
+	 * OpenSSL has no equivalent of mcrypt_get_key_size() and mcrypt_get_block_size() hence sizes stored here.
+	 *
+	 * @var array
+	 */
+	protected $supportedAlgos = array(
+		'aes-256' => array('name' => 'AES-256', 'keySize' => 32, 'blockSize' => 32),
+		'bf' => array('name' => 'BF', 'keySize' => 16, 'blockSize' => 8),
+		'des' => array('name' => 'DES', 'keySize' => 7, 'blockSize' => 8),
+		'des-ede3' => array('name' => 'DES-EDE3', 'keySize' => 21, 'blockSize' => 8), // 3 different 56-bit keys
+		'cast5' => array('name' => 'CAST5', 'keySize' => 16, 'blockSize' => 8),
+	);
+
+	/**
+	 * Supported encryption modes
+	 *
+	 * @var array
+	 */
+	protected $supportedModes = array(
+		'cbc' => 'CBC',
+	);
+
+	/**
+	 * A class to handle secure encryption and decryption of arbitrary data
+	 *
+	 *  Note that this is not just straight encryption. It also has a few other
+	 *  features in it to make the encrypted data far more secure.  Note that any
+	 *  other implementations used to decrypt data will have to do the same exact
+	 *  operations.
+	 *
+	 * Security Benefits:
+	 *
+	 * - Uses Key stretching
+	 * - Hides the Initialization Vector
+	 * - Does HMAC verification of source data
+	 *
+	 */
+
+	/**
+	 * Create an encryption key. Based on given parameters
+	 *
+	 * @param string $userKey The user key to use. This should be specific to this user.
+	 * @param string $serverKey The server key
+	 * @param string $userSuppliedKey A key from the credential (eg guid, name or tags)
+	 * @return string
+	 */
+
+	public static function makeKey($userKey, $serverKey, $userSuppliedKey) {
+		$key = hash_hmac('sha512', $userKey, $serverKey);
+		$key = hash_hmac('sha512', $key, $userSuppliedKey);
+		return $key;
+	}
+
+	/**
+	 * @var string $cipher The mcrypt cipher to use for this instance
+	 */
+	protected $cipher = '';
+
+	/**
+	 * @var int $mode The mcrypt cipher mode to use
+	 */
+	protected $mode = '';
+
+	/**
+	 * @var int $rounds The number of rounds to feed into PBKDF2 for key generation
+	 */
+	protected $rounds = 100;
```

Comment:
OK, as long as you have experimented with different values, I believe it is reasonable to leave it as is. The only thing you might want to add is a comment referring to this discussion.